client/interface.go

```diff
@@ -6,42 +6,145 @@ import (
 	"github.com/theupdateframework/notary/tuf/signed"
 )
 
-// Repository represents the set of options that must be supported over a TUF repo.
+// Reader represents the set of options that must be supported over a TUF repo for
+// reading
+type Reader interface {
```

Comment:
Wondering if there is a better name than `Reader` as that usually means an `io.Reader`...

Reply:
I am happy to name it something else.  Alternately we could just make those methods that take a TUF repo instead of having a wrapper